web/client/components/TOC/fragments/settings/General.jsx

```diff
@@ -187,12 +187,63 @@ class General extends React.Component {
                             </Col>
                         </div>
                     }
+                    <div className={"legend-options"}>
+                        <Col xs={12} className={"legend-label"}>
+                            <label key="legend-options-title" className="control-label"><Message msgId="layerProperties.legendOptions.title" /></label>
+                        </Col>
+                        <Col xs={12} sm={6} className="first-selectize">
+                            <FormGroup>
+                                <ControlLabel><Message msgId="layerProperties.legendOptions.legendWidth" /></ControlLabel>
+                                <FormControl
+                                    defaultValue={this.props.element.legendOptions && this.props.element.legendOptions.legendWidth || ''}
+                                    key="legendWidth"
+                                    type="number"
+                                    onKeyPress={(e)=> {
+                                        const number = parseInt(e.target.value + e.key, 10);
+                                        const result = Number.isInteger(number);
+                                        !result || e.key === "." ? e.preventDefault() : result && number > 1000 && e.preventDefault();
```

Comment:
Negative numbers should not be allowed, as well as too big numbers.

Reply:
Yes, this will not allow negative numbers and also the max user can feed is 1000